src/renderer/components/CanvasVisualiser/Visualiser.js

```diff
@@ -77,8 +78,19 @@ function edgesBetweenTwoNodes(a, b) {
 
 function addEdge(edge) {
   if (this.nodeExists(edge.from) && this.nodeExists(edge.to)
-        && !this.alreadyConnected(edge.from, edge.to, edge.label)) {
-    this._edges.add(Object.assign(edge, { label: '', hiddenLabel: edge.label }));
+    && !this.alreadyConnected(edge.from, edge.to, edge.label)) {
+    if (edge.options.interfaceType === interfaceTypes.SCHEMA_DESIGNER) {
+      if (edge.options.label.show) {
+        Object.assign(edge, { label: edge.label });
+      }
+      if (edge.options.arrow.show) {
+        Object.assign(edge, { arrows: { to: { enabled: true, scaleFactor: 0.5, type: edge.options.arrow.type || 'arrow' } } });
+      }
+    } else if (edge.options.interfaceType === interfaceTypes.VISUALISER) {
+      Object.assign(edge, { label: '', hiddenLabel: edge.label });
+      Object.assign(edge, { arrows: { to: { enabled: false } } });
+    }
+    this._edges.add(edge);
```

Comment:
what's happening here?

Reply:
`options.label` and `options.arrow` are only available on edge objects that need to be added to the **schema designer** and so the first condition to be met is the interface type.

Reply:
Ok, this logic should not be here though.
We have 2 components for Visualiser:
- Visualiser
- Facade
The idea of having a facade is so that the user should never interact with Visualiser directly, they do everything using Facade, so that if one day we decide to change visualisation library nothing else breaks other than `Visualiser.js`
So let's try to keep logic that is application specific, such as the concept of SCHEMA_DESIGNER or VISUALISER at least in the Facade. Ideally also Facade should know nothing about specifics of our application. 
In this specific case you're not changing the way we add the edge to the Visualiser, you're only changing the `edge` object itself, so this logic should be in https://github.com/graknlabs/workbase/blob/master/src/renderer/components/SchemaDesign/store/actions.js#L114
and other places where SchemaDesigner and Visualiser add edges to the Canvas.
If you move the logic in the appropriate components you won't even need the "interfaceType"

Reply:
Thanks for the detailed explanation.

The last commit refactors the implementation of adding and reading options on edge objects. The implementation that was previously contained within `addEdge` function is now moved to `constructEdge` function placed in `shared/SharedUtils.js`.

Reply:
Thank you, good change.
Would it make sense to have something like `constructVisualiserEdge(options)` and `constructSchemaEdge(options)` or even without options if possible (I don't have full context)?
I am thinking this because if we give it specific names it's immediately clear that we only have 2 types of base edges and we reduce the chance of passing a wrong base, also we don't repeat the same object as parameter (`base`) and improve maintainability (we only have 2 methods to change if we need to change the way we build edges) and encapsulation (all the high level components don't need to know the structure of the edge object)